### PR TITLE
docs(node-core): add document of build_blockchain_tree

### PR DESCRIPTION
**Pull request by jsvisa**

### CODE REVIEW

File: crates/node-builder/src/node.rs [hunk 1]
```diff
@@ -36,11 +36,11 @@ pub trait Node<N>: NodeTypes + Clone {
 
 /// The type that configures stateless node types, the node's primitive types.
 pub trait NodeTypes: Send + Sync + 'static {
-    /// The node's primitive types.
+    /// The node's primitive types, defining basic operations and structures.
     type Primitives: NodePrimitives;
-    /// The node's engine types.
+    /// The node's engine types, defining the interaction with the blockchain engine.
```
Comment (Rjected) [inline crates/node-builder/src/node.rs:41]:
```suggestion
    /// The node's engine types, defining the interaction with the consensus engine.
```

File: crates/node-core/src/node_config.rs [hunk 2]
```diff
@@ -419,7 +419,48 @@ impl NodeConfig {
...
+    /// ```
+    /// let tree = config.build_blockchain_tree(
+    ///     provider_factory,
+    ///     consensus,
+    ///     prune_config,
+    ///     sync_metrics_tx,
+    ///     BlockchainTreeConfig::default(),
+    ///     evm_config,
+    /// )?;
+    /// ```
```
Comment (Rjected) [inline crates/node-core/src/node_config.rs:458]: this needs either:
```
\`\`\`ignore
```
or needs real components to be initialized, or else the doctests will not work

File: crates/node-builder/src/components/payload.rs [hunk 3]
```diff
@@ -9,7 +9,7 @@ use std::future::Future;
 pub trait PayloadServiceBuilder<Node: FullNodeTypes, Pool: TransactionPool>: Send {
     /// Spawns the payload service and returns the handle to it.
     ///
-    /// The [BuilderContext] is provided to allow give to access the node's configuration.
+    /// The [BuilderContext] is provided to allow given access the node's configuration.
```
Comment (Rjected) [inline crates/node-builder/src/components/payload.rs:12]:
```suggestion
    /// The [BuilderContext] is provided to allow access to the node's configuration.
```

**Review by Rjected — CHANGES_REQUESTED** [review]
thx! i have some nits on wording and doctest validity

File: crates/node-core/src/node_config.rs (hunk 2 above)
Comment (jsvisa) [reply crates/node-core/src/node_config.rs:458]: thanks for the tips

**Review by mattsse — APPROVED** [review]
ty,
pending @Rjected

**Review by Rjected — APPROVED** [review]
looks good to me, thanks!

**Pull request merged**